### PR TITLE
Fixed line too long pep8 error for chained methods

### PR DESCRIPTION
**Pull request by MahendraNG**
Fixed pep8 issues

### CODE REVIEW

File: src/addresses.py [hunk 1]
```diff
@@ -157,126 +191,150 @@ def decodeAddress(address):
...
         if len(embeddedRipeData) == 19:
-            return status,addressVersionNumber,streamNumber,'\x00'+embeddedRipeData
+            return status, addressVersionNumber, streamNumber, '\x00' + embeddedRipeData
         elif len(embeddedRipeData) == 20:
-            return status,addressVersionNumber,streamNumber,embeddedRipeData
+            return status, addressVersionNumber, streamNumber, embeddedRipeData
         elif len(embeddedRipeData) == 18:
-            return status,addressVersionNumber,streamNumber,'\x00\x00'+embeddedRipeData
+            return status, addressVersionNumber, streamNumber, '\x00\x00'
+            + embeddedRipeData
```
Comment (PeterSurda) [inline src/addresses.py]: This is not formatted correctly.

File: src/helper_random.py [hunk 2]
```diff
@@ -1,9 +1,41 @@
...
         return os.urandom(n)
     except NotImplementedError:
         return OpenSSL.rand(n)
+
+
+def randomShuffle(population):
+    # Shuffle the sequence x in place.
+    # shuffles the elements in list in place,
+    # so they are in a random order.
+    return shuffle(population)
```
Comment (PeterSurda) [inline src/helper_random.py]: I'm not sure this works, maybe should be `random.shuffle`

File: src/helper_random.py [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+
+
+def randomChoice(population):
+    # Generate a random value from the sequence sequence.
+    # Return a k sized list of elements chosen
+    # from the population with replacement.
+    # If the population is empty, raises IndexError.
+    return random.SystemRandom().choice(population)
```
Comment (PeterSurda) [inline src/helper_random.py]: At the moment I would omit `SystemRandom()`

File: src/addresses.py [hunk 4]
```diff
@@ -157,126 +191,150 @@ def decodeAddress(address):
...
-            return 'ripetoolong',0,0,""
+            return 'ripetoolong', 0, 0, ""
         elif len(embeddedRipeData) < 4:
-            return 'ripetooshort',0,0,""
+            return 'ripetooshort', 0, 0, ""
         else:
             x00string = '\x00' * (20 - len(embeddedRipeData))
-            return status,addressVersionNumber,streamNumber,x00string+embeddedRipeData
+            return status, addressVersionNumber, streamNumber, x00string
+            +embeddedRipeData
```
Comment (PeterSurda) [inline src/addresses.py]: This isn't formatted correctly

File: src/bitmessageqt/foldertree.py [hunk 5]
```diff
@@ -22,27 +22,27 @@ class AccountMixin(object):
...
 
-    def accountColor (self):
+    def accountColor(self):
         if not self.isEnabled:
             return QtGui.QColor(128, 128, 128)
         elif self.type == self.CHAN:
             return QtGui.QColor(216, 119, 0)
         elif self.type in [self.MAILINGLIST, self.SUBSCRIPTION]:
-            return QtGui.QColor(137, 04, 177)
+            return QtGui.QColor(137, 0o4, 177)
```
Comment (PeterSurda) [inline src/bitmessageqt/foldertree.py]: ?

File: src/bitmessageqt/foldertree.py [hunk 6]
```diff
@@ -449,24 +510,34 @@ def setData(self, role, value):
...
-        return super(Ui_AddressBookWidgetItem, self).setData(role, value)    
+        return super(Ui_AddressBookWidgetItem, self).setData(role, value)
 
-    def __lt__ (self, other):
+    def __lt__(self, other):
         if (isinstance(other, Ui_AddressBookWidgetItem)):
             reverse = False
-            if self.tableWidget().horizontalHeader().sortIndicatorOrder() == QtCore.Qt.DescendingOrder:
+            if self.tableWidget().horizontalHeader(
+            ).sortIndicatorOrder() == QtCore.Qt.DescendingOrder:
```
Comment (PeterSurda) [inline src/bitmessageqt/foldertree.py]: This looks strange. Why not 
```
horizontalHeader()
.sortIndicatorOrder()
```
?

**Comment by g1itch** [comment]
So weird formatting... Did you see my variant (81b12d6)? What's wrong?

File: src/class_singleWorker.py [hunk 7]
```diff
@@ -217,41 +255,54 @@ def doPOWForMyV2Pubkey(self, hash):  # This function also broadcasts out the pub
...
+        # According to the protocol specification, the expiresTime along with
+        # the pubkey information is
+        # signed. But to be backwards compatible during the upgrade period, we
+        # shall sign not the
+        # expiresTime but rather the current time. There must be precisely a
+        # 28 day difference
+        # between the two. After the upgrade period we'll switch to signing
+        # the whole payload with the
+        # expiresTime time.
         """
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: I think this can be formatted better

File: src/class_singleWorker.py [hunk 8]
```diff
@@ -346,37 +409,52 @@ def sendOutOrStoreMyV4Pubkey(self, myAddress):
             myAddress, 'noncetrialsperbyte'))
         dataToEncrypt += encodeVarint(BMConfigParser().getint(
             myAddress, 'payloadlengthextrabytes'))
-        
+
         # When we encrypt, we'll use a hash of the data
         # contained in an address as a decryption key. This way in order to
-        # read the public keys in a pubkey message, a node must know the address
+        # read the public keys in a pubkey message, a node must know the
+        # address
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: This whole comment can be formatted better.

File: src/class_singleWorker.py [hunk 9]
```diff
@@ -410,107 +497,147 @@ def sendBroadcast(self):
...
-            # and being stored on the disk of 3rd parties. 
+            # Encrypt the broadcast with the information contained in the
+            # broadcaster's address.
+            # Anyone who knows the address can generate the private encryption
+            # key to decrypt
+            # the broadcast. This provides virtually no privacy; its purpose
+            # is to keep
+            # questionable and illegal content from flowing through the
+            # Internet connections
+            # and being stored on the disk of 3rd parties.
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: This paragraph can be formatted better

File: src/class_singleWorker.py [hunk 10]
```diff
@@ -410,107 +497,147 @@ def sendBroadcast(self):
...
+            # Sanity check. The payload size should never be larger
+            # than 256 KiB. There should
+            # be checks elsewhere in the code to not let the user try to send
+            # a message this large
+            # until we implement message continuation.
+            if len(payload) > 2 ** 18:  # 256 KiB
+                logger.critical(
+                    'This broadcast object is too large to\
+                    send. This should never happen.\
+                    Object size: %s' % len(payload))
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: I think you can use `,` instead of `%`.

File: src/class_singleWorker.py [hunk 11]
```diff
@@ -520,120 +647,176 @@ def sendBroadcast(self):
...
+            # We may or may not already have the pubkey for this toAddress.
+            #  Let's check.
             if status == 'forcepow':
-                # if the status of this msg is 'forcepow' then clearly we have the pubkey already
-                # because the user could not have overridden the message about the POW being
+                # if the status of this msg is 'forcepow' then clearly we have
+                # the pubkey already
+                # because the user could not have overridden the message about
+                # the POW being
                 # too difficult without knowing the required difficulty.
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: This paragraph can be formatted better

File: src/class_singleWorker.py [hunk 12]
```diff
@@ -520,120 +647,176 @@ def sendBroadcast(self):
...
                 pass
             elif status == 'doingmsgpow':
-                # We wouldn't have set the status to doingmsgpow if we didn't already have the pubkey
+                # We wouldn't have set the status to doingmsgpow if we didn't
+                # already have the pubkey
                 # so let's assume that we have it.
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: this paragraph can be formatted better

File: src/class_singleWorker.py [hunk 13]
```diff
@@ -520,120 +647,176 @@ def sendBroadcast(self):
...
                     status = 'doingmsgpow'
-                    # mark the pubkey as 'usedpersonally' so that we don't delete it later. If the pubkey version
-                    # is >= 4 then usedpersonally will already be set to yes because we'll only ever have 
+                    # mark the pubkey as 'usedpersonally'
+                    # so that we don't delete it later
+                    # . If the pubkey version
+                    # is >= 4 then usedpersonally
+                    # will already be set to yes
+                    # because we'll only ever have
                     # usedpersonally v4 pubkeys in the pubkeys table.
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: this paragraph can be formatted better

File: src/class_singleWorker.py [hunk 14]
```diff
@@ -520,120 +647,176 @@ def sendBroadcast(self):
...
+                            # inventory.
+                            # If we have it we'll need to decrypt it and put
+                            # it in
+                            # the pubkeys table.
+
+                            # The decryptAndCheckPubkeyPayload function expects
+                            # that the shared.neededPubkeys
+                            # dictionary already contains the toAddress and
+                            # cryptor object associated with
                             # the tag for this toAddress.
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: these 2 paragraphs can be formatted better

File: src/class_singleWorker.py [hunk 15]
```diff
@@ -772,91 +1051,166 @@ def sendMsg(self):
...
-            # until we implement message continuation. 
-            if len(encryptedPayload) > 2 ** 18: # 256 KiB
-                logger.critical('This msg object is too large to send. This should never happen. Object size: %s' % len(encryptedPayload))
+
+            # Sanity check. The encryptedPayload size
+            # should never be larger than
+            # 256 KiB. There should
+            # be checks elsewhere in the code to not
+            # let the user try to send a message this large
+            # until we implement message continuation.
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: This paragraph can be formatted better

File: src/class_singleWorker.py [hunk 16]
```diff
@@ -880,111 +1234,189 @@ def requestPubKey(self, toAddress):
...
-            # of the client then we have to put it in now. 
-            privEncryptionKey = hashlib.sha512(hashlib.sha512(encodeVarint(addressVersionNumber)+encodeVarint(streamNumber)+ripe).digest()).digest()[:32] # Note that this is the first half of the sha512 hash.
-            tag = hashlib.sha512(hashlib.sha512(encodeVarint(addressVersionNumber)+encodeVarint(streamNumber)+ripe).digest()).digest()[32:] # Note that this is the second half of the sha512 hash.
+            # If the user just clicked 'send' then
+            #  the tag (and other information)
+            #  will already
+            # be in the neededPubkeys dictionary.
+            # But if we are recovering from a restart
+            # of the client then we have to put it in now.
+            # Note that this is the first half of the sha512 hash.
```
Comment (PeterSurda) [inline src/class_singleWorker.py]: This paragraph can be formatted better

File: src/depends.py [hunk 17]
```diff
@@ -4,14 +4,17 @@
 import os
 import pyelliptic.openssl
 
-#Only really old versions of Python don't have sys.hexversion. We don't support
-#them. The logging module was introduced in Python 2.3
+# Only really old versions of Python don't have sys.hexversion. We don't
+# support
+# them. The logging module was introduced in Python 2.3
```
Comment (PeterSurda) [inline src/depends.py]: this paragraph can be formatted better

File: src/depends.py [hunk 18]
```diff
@@ -22,28 +25,43 @@
 logger.addHandler(handler)
 logger.setLevel(logging.ERROR)
 
-#We need to check hashlib for RIPEMD-160, as it won't be available if OpenSSL is
-#not linked against or the linked OpenSSL has RIPEMD disabled.
+# We need to check hashlib for RIPEMD-160, as it won't be
+# available if OpenSSL # is
+# not linked against or the linked OpenSSL has RIPEMD disabled.
```
Comment (PeterSurda) [inline src/depends.py]: this paragraph can be formatted better

File: src/depends.py [hunk 19]
```diff
@@ -53,40 +71,53 @@ def check_sqlite():
...
-            #first version that was included with Python.
+                compile_options = ', '.join(
+                    map(lambda row: row[0],
+                        conn.execute('PRAGMA compile_options;')))
+                logger.info(
+                    'SQLite Library Compile Options: ' +
+                    compile_options)
+            # There is no specific version requirement as yet, so we just use
+            # the
+            # first version that was included with Python.
```
Comment (PeterSurda) [inline src/depends.py]: this paragraph can be formatted better

File: src/depends.py [hunk 20]
```diff
@@ -123,83 +154,126 @@ def check_openssl():
...
         return False
     logger.info('pythondialog Package Version: ' + dialog.__version__)
     dialog_util_version = dialog.Dialog().cached_backend_version
-    #The pythondialog author does not like Python2 str, so we have to use
-    #unicode for just the version otherwise we get the repr form which includes
-    #the module and class names along with the actual version.
+    # The pythondialog author does not like Python2 str, so we have to use
+    # unicode for just the version otherwise we get the repr form which
+    # includes
+    # the module and class names along with the actual version.
```
Comment (PeterSurda) [inline src/depends.py]: this paragraph can be formatted better

File: src/depends.py [hunk 21]
```diff
@@ -208,58 +282,86 @@ def check_msgpack():
...
     if verbose:
         logger.setLevel(logging.INFO)
 
     has_all_dependencies = True
 
-    #Python 2.7.3 is the required minimum. Python 3+ is not supported, but it is
-    #still useful to provide information about our other requirements.
+    # Python 2.7.3 is the required minimum. Python 3+ is not supported, but it
+    # is
+    # still useful to provide information about our other requirements.
```
Comment (PeterSurda) [inline src/depends.py]: this paragraph can be formatted better

**Comment by PeterSurda** [comment]
@g1itch Feel free to comment on the formatting. Your PR was too big and I would like Mahendra to focus on code quality exclusively at the beginning.

**Comment by g1itch** [comment]
@PeterSurda, I didn't plan to make PR from that feature as it still not finished. But with such formatting I expect heavy pain when rebase it. Some of that "formatting" changes my introduce a problems in future (like `"... {}".format` for python3) so it must be tested carefully.

There is also a tons of well formatted python code here on github. I'd suggest to find some good examples instead of trying to satisfy codacy by touch.

**Comment by PeterSurda** [comment]
@g1itch  Ok, then which parts do you recommend focusing on?

**Comment by PeterSurda** [comment]
@g1itch and what do you suggest instead of  `"... {}".format`?

File: src/bitmessageqt/foldertree.py (hunk 6 above)
Comment (g1itch) [reply src/bitmessageqt/foldertree.py]: I have this file also:

https://github.com/g1itch/PyBitmessage/commit/57a679812d130bfbf6ef66cd2f2d159e1c6d0b9e

```python
            reverse = QtCore.Qt.DescendingOrder == \
                self.tableWidget().horizontalHeader().sortIndicatorOrder()
```

**Comment by g1itch** [comment]
Also an interesting reading: https://www.python.org/dev/peps/pep-0008/

**Comment by g1itch** [comment]
Let's begin here: https://guides.github.com/introduction/flow/

**Pull request closed without merging**